megamek/src/megamek/common/Crew.java

```diff
@@ -1354,10 +1385,64 @@ public boolean doConsoleRoleSwap() {
         return false;
     }
 
+    //region extraData
+    public void initializeExtraDataAsHashMap(int crewNumber) {
+        this.extraData = new HashMap<>(crewNumber);
+        for (int i = 0; i < crewNumber; i++) {
+            this.extraData.put(i, new HashMap<>());
+        }
+    }
+
+    public void initializeExtraDataAsHashMap() {
+        this.extraData = new HashMap<>();
+    }
+
+    public static Map<Integer, Map<String, String>> getNewInitializedExtraDataAsHashMap(int crewNumber) {
+        Map<Integer, Map<String, String>> extraData = new HashMap<>(crewNumber);
+        for (int i = 0; i < crewNumber; i++) {
+            extraData.put(i, new HashMap<>());
+        }
+
+        return extraData;
+    }
+
+    public void setExtraData(Map<Integer, Map<String, String>> extraData) {
+        this.extraData = extraData;
+    }
+
+    public void setExtraDataForCrewMember(int crewIndex, Map<String, String> dataMap) {
+        this.extraData.put(crewIndex, dataMap);
+    }
+
+    public void addExtraDataDataPoint(int crewIndex, String key, String value) {
+        this.extraData.get(crewIndex).put(key, value);
```

Comment:
Should this create the hashmap if the crew member's slot doesn't exist?

Reply:
Yes, it should. Fixing that.

Reply:
If we expect just 2 or 3 items per crew member, set the default capacity of the inner hash maps to 2 or 4.